### DISCUSSION TITLE
Half-baked idea, since Brunch modifies the Cros kernel, what would it take to use the Microsoft WSL2 kernel instead?

### DISCUSSION THREAD

**Discussion by Warrentheo** [sebanc/brunch#1348] [Ideas]
This is probably along the same lines as my previous Nvidia GPU question, and will probably be more pain than it is worth, but I think this one is at least closer to being possible, and therefore is at least worth asking the question, so here it goes...

Windows 11 has brought with it the "Windows Subsystem for Linux" version 2, that now has nearly 100% full Linux support, including even supporting full graphics driver support, offering near native support for graphics applications now, and I can personally confirm it working very well...
It also currently has support for the "Windows Subsystem for Android with Amazon Appstore", and I can confirm it also working, including in games at near native speeds...

Some things of interest to this:

- The current version of the Microsoft Kernel as of writing this is "5.10.74.3-microsoft-standard-WSL2" and "5.10.43-2-windows-subsystem-for-android (eng.latteu.20211111.21.0252)" with a similar named kernel under WSL as well...  I don't know for sure, but I think these are currently fairly static, but expected to start updating automatically in the future with this link: [ms-windows-store://pdp/?ProductId=9p9tqf7mrm4r](url), and I think all WSL2 based distros are expected to share this same installation of the kernel...  This is probably the biggest reason why this would not be easy to get working...

- WSL2 creates a fake hard drive using a VHD(x) file, like for Hyper-V...  As far as I know, this can be repartitioned as needed by Brunch...

- The CPU-Z app running under WSA reports the GPU as **"Microsoft (Google Inc. (NVIDIA))"** with a GPU Renderer **"Android Emulator OpenGL ES Translator"** on my system because of me installing OpenGApps, which adds Google Services to the WSA in a way where it thinks that I have a Google Pixel 5...  I have attached screenshots below

- WSL2 GPU support currently requires special drivers, they can be found at these locations (don't know about Intel support):
[https://www.amd.com/en/support/kb/release-notes/rn-rad-win-wsl-support](url)
[https://developer.nvidia.com/cuda/wsl/download](url)

- Google Play Store and Google Play Services for WSA can be found here: [https://github.com/WSA-Community/WSAGAScript](url).  This was a great start for figuring out a decent amount of this...

- I have attached some screenshots from my WSL2 and WSA installations for context, and I am willing to attach any other requested files from my installs as needed to help answer the questions...

![WSA 1  SOC](https://user-images.githubusercontent.com/5157428/146704442-664a7ea7-ab93-49d0-b804-7e7f054f864e.png)
![WSA 2  Device](https://user-images.githubusercontent.com/5157428/146704445-adef40b4-1093-4a11-83d2-572512b07ac4.png)
![WSA 3  System](https://user-images.githubusercontent.com/5157428/146704451-2330ff99-34fd-42f2-bf33-6472ad76c9c9.png)
![WSA 4  Battery](https://user-images.githubusercontent.com/5157428/146704454-2371a2bc-2dc2-40d6-bf16-b819f3ab6835.png)
![WSL2 1  Kernel](https://user-images.githubusercontent.com/5157428/146704580-28356010-535c-4527-a285-38b88ff515df.png)
![WSL2 2  Swap Drive](https://user-images.githubusercontent.com/5157428/146704583-a8013153-d1f2-428f-8353-58ee3756c904.png)
![WSL2 3  Root](https://user-images.githubusercontent.com/5157428/146704589-fd00f8b0-04c6-40c6-945e-4b0c271e6688.png)

**Comment by ycavan** [comment]
What is the purpose of this?  if you want to use an NVidia video card, install the drivers using Arch linux under brioche... If you want to use virtual machines, use qemu... I'm trying to understand the use case for installing Microsoft's Hypervisor when we have KVM?

**Reply by Warrentheo** [reply]
Using ChromeOS as the guest VM, not the host...  There currently is no Nvidia Drivers for running ChromeOS bare metal, but there is when running inside of a WSL instance now...
